src/mono/mono/metadata/seq-points-data.c

```diff
@@ -414,7 +414,7 @@ mono_seq_point_data_read (SeqPointData *data, char *path)
 		fclose(f);
 		return FALSE;
 	}
-	g_assert (len == fsize || (len < fsize && feof(f)));
+	g_assert (len == fsize || (len < GLONG_TO_ULONG(fsize) && feof(f)));
```

Comment:
`fsize` could be larger than what `size_t` could hold.

Reply:
size_t normally aligns to max register size of the machine, so 4 or 8 bytes, max of long is undefined, but it should be at least 4 bytes and signed, so in theory on 32-bit system (and on platforms that defines long to be > 4 bytes), it could be true, but not sure that has any implementation on platforms we support. So it could be incorrect either way, but I'm fine leaving it as is and let the compiler to cast it as needed.